include/seqan3/range/view/translate_join.hpp

```diff
@@ -0,0 +1,405 @@
+    constexpr view_translate_join(view_translate_join && rhs)                  noexcept = default; //!< Defaulted.
+    constexpr view_translate_join & operator=(view_translate_join const & rhs) noexcept = default; //!< Defaulted.
+    constexpr view_translate_join & operator=(view_translate_join && rhs)      noexcept = default; //!< Defaulted.
+    ~view_translate_join()                                                     noexcept = default; //!< Defaulted.
+
+    /*!\brief Construct from another view.
+     * \param[in] _urange The underlying range (of ranges).
+     * \param[in] _tf The frames that should be used for translation.
+     */
+    view_translate_join(urng_t _urange, translation_frames const _tf = translation_frames::SIX_FRAME)
+        : urange{std::move(_urange)}, tf{_tf}
+    {
+        if ((_tf & translation_frames::FWD_FRAME_0) == translation_frames::FWD_FRAME_0)
+            selected_frames.push_back(translation_frames::FWD_FRAME_0);
+        if ((_tf & translation_frames::FWD_FRAME_1) == translation_frames::FWD_FRAME_1)
+            selected_frames.push_back(translation_frames::FWD_FRAME_1);
+        if ((_tf & translation_frames::FWD_FRAME_2) == translation_frames::FWD_FRAME_2)
+            selected_frames.push_back(translation_frames::FWD_FRAME_2);
+        if ((_tf & translation_frames::REV_FRAME_0) == translation_frames::REV_FRAME_0)
+            selected_frames.push_back(translation_frames::REV_FRAME_0);
+        if ((_tf & translation_frames::REV_FRAME_1) == translation_frames::REV_FRAME_1)
+            selected_frames.push_back(translation_frames::REV_FRAME_1);
+        if ((_tf & translation_frames::REV_FRAME_2) == translation_frames::REV_FRAME_2)
+            selected_frames.push_back(translation_frames::REV_FRAME_2);
+    }
+
+    /*!\brief Construct from another range.
+     * \param[in] _urange The underlying range (of ranges).
+     * \param[in] _tf The frames that should be used for translation.
+     */
+    template <typename rng_t>
+    //!\cond
+        requires !std::Same<remove_cvref_t<rng_t>, view_translate_join> &&
+                 std::ranges::ViewableRange<rng_t> &&
+                 std::Constructible<urng_t, ranges::ref_view<std::remove_reference_t<rng_t>>>
+    //!\endcond
+    view_translate_join(rng_t && _urange, translation_frames const _tf = translation_frames::SIX_FRAME)
+     : view_translate_join{std::view::all(std::forward<rng_t>(_urange)), _tf}
+    {}
+    //!\}
+
+    /*!\name Iterators
+     * \{
+     */
+    /*!\brief Returns an iterator to the first element of the container.
+     * \returns Iterator to the first element.
+     *
+     * If the container is empty, the returned iterator will be equal to end().
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    iterator begin() noexcept
+    {
+        return {*this, 0};
+    }
+
+    //!\overload
+    const_iterator begin() const noexcept
```

Comment:
```suggestion
    const_iterator begin() const noexcept
        requires ConstIterableRange<urng_t>
```

Also for the other const-qualified member functions.